[WPE] WPE Platform: Unset WPE_DRM_DEVICE for Tests using WPEMockPlatformTest

lauromoura opened the pull request:
#### 6339f3f4305730fdd43ec643cb8a720b7d5f53a1
<pre>
[WPE] WPE Platform: Unset WPE_DRM_DEVICE for Tests using WPEMockPlatformTest
<a href="https://bugs.webkit.org/show_bug.cgi?id=300535">https://bugs.webkit.org/show_bug.cgi?id=300535</a>

Reviewed by Carlos Garcia Campos.

wpe_display_get_drm_device gives higher precedence to the override
device over the one provided by the actual platform Display class, while
in the Mock plaform we want to use the mocked classes.

This commit ensures we unset it for the Mock tests.

* Tools/TestWebKitAPI/glib/WPEPlatform/WPEPlatformTestMain.cpp:
(main):

Canonical link: <a href="https://commits.webkit.org/301390@main">https://commits.webkit.org/301390@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7c776ec7cc24e3bcdc9a1677199342e5f3b09b0b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125724 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/45386 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/36137 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/132587 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77607 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/127595 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/46071 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53945 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95783 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/63902 "An unexpected error occured. Recent messages:Printed configuration") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128672 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36832 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/112426 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/76275 "Found 1 new API test failure: WPE/TestWebKitWebContext:/webkit/WebKitWebContext/no-web-process-leak (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35734 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/30612 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/76058 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106610 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30825 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/135267 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52513 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/40271 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/104248 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52961 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108638 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103976 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26498 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/49339 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27652 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/49786 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/52408 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51757 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/55107 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53452 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Tools/TestWebKitAPI/Tests/WPEPlatform/TestDisplay.cpp:
```diff
@@ -302,10 +304,16 @@ void beforeAll()
+
+    if (const char* prev = g_getenv("WPE_DRM_DEVICE"))
+        s_previousDRMDeviceOverride = prev;
+    g_unsetenv("WPE_DRM_DEVICE");
```
I think we can do this in WPEPlatformTestMain.cpp

carlosgcampos commented on Tools/TestWebKitAPI/Tests/WPEPlatform/TestDisplay.cpp:
```diff
@@ -302,10 +304,16 @@ void beforeAll()
 void afterAll()
 {
+    if (s_previousDRMDeviceOverride)
+        g_setenv("WPE_DRM_DEVICE", s_previousDRMDeviceOverride->c_str(), TRUE);
```
And we don't need to set it again, because the test program finishes right after this.

lauromoura committed:
[WPE] WPE Platform: Unset WPE_DRM_DEVICE for Tests using WPEMockPlatformTest
https://bugs.webkit.org/show_bug.cgi?id=300535

Reviewed by Carlos Garcia Campos.

wpe_display_get_drm_device gives higher precedence to the override
device over the one provided by the actual platform Display class, while
in the Mock plaform we want to use the mocked classes.

This commit ensures we unset it for the Mock tests.

* Tools/TestWebKitAPI/glib/WPEPlatform/WPEPlatformTestMain.cpp:
(main):

Canonical link: https://commits.webkit.org/301390@main
